lib/open_food_network/enterprise_injection_data.rb

```diff
@@ -1,7 +1,7 @@
 module OpenFoodNetwork
   class EnterpriseInjectionData
     def active_distributors
-      @active_distributors ||= Enterprise.distributors_with_active_order_cycles
+      @active_distributors ||= Enterprise.distributors_with_active_order_cycles.andand.ready_for_checkout
```

Comment:
I found three usages of this:

- app/helpers/shared_helper.rb
- app/serializers/api/enterprise_serializer.rb
- app/views/json/partials/_hub.rabl

All of them call the `include?` method on that collection to know if a certain enterprise is active. That sounds like a really poor way of getting that attribute. The computation time will grow linearly with the number of active enterprises.

So this fixes the issue, but it also points to a possible performance problem.